langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/utils/ArrayUtils.java

```diff
@@ -121,6 +122,10 @@ public static BArray createEmptyArrayFromTuple(BArray arr) {
         }
         // Create an array of one type if the member-type-descriptors are the same
         if (isSameType) {
+            // Create an array with never type as element type if the `sameType` is null
+            if (sameType == null) {
+                return ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_NEVER));
```

Comment:
Can we add this before line 109?
 if (tupleTypes.isEmpty()) {
             return ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_NEVER));
        }